input/collector.go

```diff
@@ -29,39 +31,40 @@ type Collector interface {
 	Addr() net.Addr
 }
 
-// TCPCollector represents a network collector that accepts and handler TCP connections.
+// TCPCollector represents a network collector that accepts TCP packets.
 type TCPCollector struct {
-	iface  string
-	parser *RFC5424Parser
-
+	iface     string
 	addr      net.Addr
 	tlsConfig *tls.Config
+	format    string
 }
 
 // UDPCollector represents a network collector that accepts UDP packets.
 type UDPCollector struct {
 	addr   *net.UDPAddr
-	parser *RFC5424Parser
+	format string
+}
+
+func (s *TCPCollector) Addr() net.Addr {
+	return s.addr
 }
 
 // NewCollector returns a network collector of the specified type, that will bind
 // to the given inteface on Start(). If config is non-nil, a secure Collector will
 // be returned. Secure Collectors require the protocol be TCP.
-func NewCollector(proto, iface string, tlsConfig *tls.Config) Collector {
-	parser := NewRFC5424Parser()
+func NewCollector(proto string, iface string, tlsConfig *tls.Config, format string) Collector {
```

Comment:
We should make `format` typed (an enum?) otherwise the caller can pass in a unsupported string and get no error. Other wise `NewCollector` could also return an `error`.